ghcide/test/exe/Main.hs

```diff
@@ -5464,8 +5467,23 @@ unitTests = do
             actualOrder <- liftIO $ readIORef orderRef
 
             liftIO $ actualOrder @?= reverse [(1::Int)..20]
+     , testCase "timestamps have millisecond resolution" $ do
+         resolution_us <- findResolution_us 1
+         let msg = printf "Timestamps do not have millisecond resolution: %dus" resolution_us
+         assertBool msg (resolution_us <= 1000)
```

Comment:
This is very much OS and file format specific. Docker containers and Mac machines often don't have this, even when using precise functions.

Reply:
This is a test that whatever we are using, file timestamps can be read with a certain precision. The precision will be OS specific, but without millisecond precision I expect that there will be trouble - other tests will fail mysteriously.
The test passes in all the platforms so far, including my Mac.

Reply:
Fair enough - I guess this is a canary for the test suite, rather than a requirement. In Shake I often have to introduce file system modification resolution sleeps to get the tests to pass.